Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.Tests.ps1

```diff
@@ -0,0 +1,85 @@
+	
+    Write-Debug("folderIdTestRootOriginal[" +  $ishFolderTestRootOriginal.IshFolderRef + "] folderTypeTestRootOriginal[" + $folderTypeTestRootOriginal + "]")
+	$global:ishBackgroundTaskCmdlet = Add-IshFolder -IShSession $ishSession -ParentFolderId $folderIdTestRootOriginal -FolderType $folderTypeTestRootOriginal -FolderName $cmdletName -OwnedBy $ownedByTestRootOriginal -ReadAccess $readAccessTestRootOriginal
+	$ishFolderTopic = Add-IshFolder -IshSession $ishSession -ParentFolderId($global:ishBackgroundTaskCmdlet.IshFolderRef) -FolderType ISHModule -FolderName "Topic" -OwnedBy $ownedByTestRootOriginal -ReadAccess $readAccessTestRootOriginal
+
+	$ishTopicMetadata = Set-IshMetadataField -IshSession $ishSession -Name "FTITLE" -Level Logical -Value "Topic $timestamp" |
+						    Set-IshMetadataField -IshSession $ishSession -Name "FAUTHOR" -Level Lng -ValueType Element -Value $ishUserAuthor |
+						    Set-IshMetadataField -IshSession $ishSession -Name "FSTATUS" -Level Lng -ValueType Element -Value $ishStatusDraft
+    $ishObjectTopic1 = Add-IshDocumentObj -IshSession $ishSession -FolderId $ishFolderTopic.IshFolderRef -IshType ISHModule -Lng $ishLng -Metadata $ishTopicMetadata -FileContent $ditaTopicFileContent
+    $ishObjectTopic1 = Add-IshDocumentObj -IshSession $ishSession -FolderId $ishFolderTopic.IshFolderRef -IshType ISHModule -Lng $ishLng -Metadata $ishTopicMetadata -FileContent $ditaTopicFileContent
+
+	$ishObjects = $ishFolderTopic | Get-IshFolderContent -IshSession $ishSession
+	Context "Add-IshBackgroundTask IshObjects passed via pipeline" {
+		$ishBackgroundTaskIshObjectsPipeline = $ishObjects | Add-IshBackgroundTask -IshSession $ishSession -EventType $eventType
+		It "Verify object properties returned by Add-IshBackgroundTask" {
+			$ishBackgroundTaskIshObjectsPipeline.Count | Should BeExactly 1
+			$ishBackgroundTaskIshObjectsPipeline.status_task_element | Should BeExactly "VBACKGROUNDTASKSTATUSPENDING"
+			$ishBackgroundTaskIshObjectsPipeline.EventType | Should BeExactly $eventType
+			$ishBackgroundTaskIshObjectsPipeline.userid | Should BeExactly $ishSession.IshUserName
+		}
+		It "Pipe returned BackgroundTask to Get-IshBackgroundTask"{
+			$ishBackgroundTask = $ishBackgroundTaskIshObjectsPipeline | Get-IshBackgroundTask -IshSession $ishSession
+			$ishBackgroundTask.Count | Should BeExactly 1
+			$ishBackgroundTask.EventType | Should BeExactly $eventType
+			$ishBackgroundTask.userid | Should BeExactly $ishSession.IshUserName
+		}
+	}
+
+	Context "Add-IshBackgroundTask IshObjects passed as parameter" {
+		$ishBackgroundTaskIshObjectsParameter = Add-IshBackgroundTask -IshSession $ishSession -EventType $eventType -IshObject $ishObjects
+		It "Verify object properties returned by Add-IshBackgroundTask" {
+			$ishBackgroundTaskIshObjectsParameter.Count | Should BeExactly 1
+			$ishBackgroundTaskIshObjectsParameter.status_task_element | Should BeExactly "VBACKGROUNDTASKSTATUSPENDING"
```

Comment:
Don't check on the status of the background task unless you stop the background task service this will not be correct

Reply:
Done